gettsim/pre_processing/policy_for_date.py

```diff
@@ -49,61 +50,146 @@
 
 
 def get_policies_for_date(policy_date, groups="all"):
-    if isinstance(policy_date, str):
-        policy_date = pd.to_datetime(policy_date)
+    """Get the state of the policy system for a particular date.
+
+    The function returns time dependent policy reforms (functions) as well as the
+    policy parameters for the current policy system.
+
+    Parameters
+    ----------
+    policy_date : datetime.date, str, int
+                  The date for which the policy system is set up.
+    groups : list, str
+             The group or a list of groups which parameters are loaded. Default is
+             all parameters
+
+    Returns
+    -------
+    params_dict : dict
+                Dictionary of parameters grouped in policy system compartments given
+                in groups.
+    policy_func_dict : dict
+                Dictionary of time dependent policy reforms. Keys are the
+                variable names they create.
+
+
+    """
+    # Check policy date for correct format and transfer to datetime.date
+    policy_date = check_date(policy_date)
+
+    # Check groups argument for correct format and transfer to list.
+    group_list = check_groups(groups)
+
+    params_dict = {}
+
+    for group in group_list:
+        tax_data = load_data(policy_date, group)
+
+        # Align paramters for e.g. piecewise polynomial functions
+        params_dict[group] = align_parameters(tax_data)
+
+    policy_func_dict = load_reforms_for_date(policy_date)
+
+    return params_dict, policy_func_dict
+
+
+def align_parameters(tax_data):
+    """Check if parameters are stored in implicit structures and align to general
+    structure.
+
+    Parameters
+    ----------
+    tax_data : dict
+            Loaded raw tax data.
+
+    Returns
+    -------
+
+    """
+    for param in tax_data:
+        if type(tax_data[param]) == dict:
+            if "type" in tax_data[param]:
+                if tax_data[param]["type"].startswith("piecewise"):
+                    if "progressionsfaktor" in tax_data[param]:
+                        if tax_data[param]["progressionsfaktor"]:
+                            tax_data[param] = add_progressionsfaktor(
+                                tax_data[param], param
+                            )
+                    tax_data[param] = get_piecewise_parameters(
+                        tax_data[param],
+                        param,
+                        func_type=tax_data[param]["type"].split("_")[1],
+                    )
+            for key in ["type", "progressionsfaktor"]:
+                tax_data[param].pop(key, None)
+
+    return tax_data
+
+
+def check_groups(groups):
```

Comment:
I find one `return out` and multiple `out = ...` more readable than multiple returns.

Call it `parse_parameter_groups`.

On errors, you can provide the full list of possible groups which helps the user to detect spelling errors and all possible options.